cli/pkg/kctrl/cmd/package/installed/pause_or_kick.go

```diff
@@ -0,0 +1,258 @@
+	err = o.ui.AskForConfirmation()
+	if err != nil {
+		return err
+	}
+
+	return o.pause(client)
+}
+
+func (o *PauseOrKickOptions) Kick() error {
+	if len(o.Name) == 0 {
+		return fmt.Errorf("Expected package install name to be non empty")
+	}
+
+	client, err := o.depsFactory.KappCtrlClient()
+	if err != nil {
+		return err
+	}
+
+	_, err = client.PackagingV1alpha1().PackageInstalls(
+		o.NamespaceFlags.Name).Get(context.Background(), o.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	o.ui.PrintLinef("Triggering reconciliation for package install '%s' in namespace '%s'", o.Name, o.NamespaceFlags.Name)
+	err = o.ui.AskForConfirmation()
+	if err != nil {
+		return err
+	}
+
+	err = o.pause(client)
+	if err != nil {
+		return err
+	}
+
+	err = o.unpause(client)
+	if err != nil {
+		return err
+	}
+
+	if o.WaitFlags.Enabled {
+		return o.waitForPackageInstallReconciliation(client)
+	}
+
+	return nil
+}
+
+func (o *PauseOrKickOptions) pause(client kcclient.Interface) error {
+	pausePatch := []map[string]interface{}{
+		{
+			"op":    "add",
+			"path":  "/spec/paused",
+			"value": true,
+		},
+	}
+
+	patchJSON, err := json.Marshal(pausePatch)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.PackagingV1alpha1().PackageInstalls(o.NamespaceFlags.Name).Patch(context.Background(), o.Name, types.JSONPatchType, patchJSON, metav1.PatchOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *PauseOrKickOptions) unpause(client kcclient.Interface) error {
+	unpausePatch := []map[string]interface{}{
+		{
+			"op":   "remove",
+			"path": "/spec/paused",
+		},
+	}
+
+	patchJSON, err := json.Marshal(unpausePatch)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.PackagingV1alpha1().PackageInstalls(o.NamespaceFlags.Name).Patch(context.Background(), o.Name, types.JSONPatchType, patchJSON, metav1.PatchOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// waitForPackageInstallReconciliation waits until the package get installed successfully or a failure happen
+// TODO Move reconciliation to a common place for create-or-update and pause-or-kick
+func (o *PauseOrKickOptions) waitForPackageInstallReconciliation(client kcclient.Interface) error {
+	o.statusUI.PrintMessagef("Waiting for PackageInstall reconciliation for '%s'", o.Name)
+	msgsUI := cmdcore.NewDedupingMessagesUI(cmdcore.NewPlainMessagesUI(o.ui))
+	description := getPackageInstallDescription(o.Name, o.NamespaceFlags.Name)
+
+	appStatusTailErrored := false
+	tailAppStatusOutput := func(tailErrored *bool) {
+		appWatcher := cmdapp.NewAppTailer(o.NamespaceFlags.Name, o.Name, o.ui, client, cmdapp.AppTailerOpts{
+			IgnoreNotExists: true,
+		})
+
+		err := appWatcher.TailAppStatus()
+		if err != nil {
+			o.statusUI.PrintMessagef("Error tailing or reconciling app: %s", err.Error())
+			*tailErrored = true
+		}
+	}
+	go tailAppStatusOutput(&appStatusTailErrored)
```

Comment:
*question:* I am just trying to think why we need to do this in parallel. Maybe we could call TailAppStatus() first and if it errors out then do the polling like we used to, otherwise return from there itself?

Reply:
That means we stop execution till TailAppStatus exits. 
Polling in the background means we are ensuring that the package install isn't erroring out at the same time. 

One case i recall where I recall this was useful is when a wrong version is supplied

Reply:
> the only scenario where we should go ahead with the polling is when app is not found

I could be wrong here. Need to think this throughly.

Reply:
Resolving it based on slack discussion:
TailAppStatus doesn't error out in all the scenarios.